src/main/java/seedu/tinner/model/reminder/Reminder.java

```diff
@@ -0,0 +1,98 @@
+package seedu.tinner.model.reminder;
+
+import static seedu.tinner.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.tinner.model.company.CompanyName;
+import seedu.tinner.model.role.Deadline;
+import seedu.tinner.model.role.RoleName;
+
+public class Reminder {
+
+    private final CompanyName companyName;
+    private final RoleName roleName;
+    private final Deadline deadline;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Reminder(CompanyName companyName, RoleName roleName, Deadline deadline) {
```

Comment:
Thinking if it will be clearer for reminder to include the status?
For example, Google, Software Engineering, Applying by 03 April etc.